crates/subspace-node/src/commands/run/consensus.rs

```diff
@@ -584,7 +596,13 @@ pub(super) fn create_consensus_chain_configuration(
         chain_spec: Box::new(chain_spec),
         informant_output_format: OutputFormat { enable_color },
     };
-    let consensus_chain_config = Configuration::from(consensus_chain_config);
+    let mut consensus_chain_config = Configuration::from(consensus_chain_config);
+    if fast_sync {
+        consensus_chain_config.network.sync_mode = SyncMode::LightState {
```

Comment:
There do not seem to be any TODOs that this is a hack that should go away once we don't have to do this anymore as described in https://github.com/paritytech/polkadot-sdk/issues/4407